packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -150,18 +150,20 @@ const styleArg = ({ theme, ownerState }: { theme: Theme; ownerState: OwnerState
       backgroundColor: (theme.vars || theme).palette.primary.dark,
     },
   },
-  [`&.${pickersDayClasses.selected}`]: {
-    color: (theme.vars || theme).palette.primary.contrastText,
-    backgroundColor: (theme.vars || theme).palette.primary.main,
-    fontWeight: theme.typography.fontWeightMedium,
-    transition: theme.transitions.create('background-color', {
-      duration: theme.transitions.duration.short,
-    }),
-    '&:hover': {
-      willChange: 'background-color',
-      backgroundColor: (theme.vars || theme).palette.primary.dark,
+  ...(!ownerState.outsideCurrentMonth && {
```

Comment:
Fixes the following issue:
![Screenshot 2023-05-31 at 17 40 34](https://github.com/mui/mui-x/assets/4941090/0215dd50-a083-4c76-ae5e-b23444b3b6d4)